### PR TITLE
Add x509svid.Verify() unit test

### PR DESCRIPTION
**Pull request by martincapello**
This PR adds the missing unit test for x509svid.Verify() and x509svid.ParseAndVerify().

### CODE REVIEW

File: v2/svid/x509svid/verify_test.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	return []*x509.Certificate{&c}
+}
+
+func setIsCA(cert *x509.Certificate) []*x509.Certificate {
+	c := *cert
+	c.IsCA = true
+	return []*x509.Certificate{&c}
+}
+
+func setKeyUsage(cert *x509.Certificate, ku x509.KeyUsage) []*x509.Certificate {
```
Comment (azdagron) [inline v2/svid/x509svid/verify_test.go]: nit: appendKeyUsage?

File: v2/internal/test/ca.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	serial := NewSerial(tb)
+
+	uriSAN, err := url.Parse(spiffeID)
+	require.NoError(tb, err)
+
+	key := NewEC256Key(tb)
+	tmpl := &x509.Certificate{
+		SerialNumber: serial,
+		Subject: pkix.Name{
+			CommonName: fmt.Sprintf("CA %x", serial),
```
[the remark this answers is not in the document]
Comment (martincapello) [reply v2/internal/test/ca.go]: Should I fix this in v1 as well?

File: v2/internal/test/ca.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+		key:    key,
+	}
+}
+
+func (ca *CA) CreateX509SVID(spiffeID string) ([]*x509.Certificate, crypto.Signer) {
+	cert, key := CreateX509SVID(ca.tb, ca.cert, ca.key, spiffeID)
+	return append([]*x509.Certificate{cert}, ca.chain(false)...), key
+}
+
+func (ca *CA) chain(includeRoot bool) []*x509.Certificate {
```
[the remark this answers is not in the document]
Comment (martincapello) [reply v2/internal/test/ca.go]: Should I fix this in v1 as well?

File: v2/internal/test/ca.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (azdagron) [reply v2/internal/test/ca.go]: Sure, but in a separate PR.

File: v2/internal/test/ca.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (azdagron) [reply v2/internal/test/ca.go]: Sure, but in a separate PR.

**Pull request merged**